pkg/cloudprovider/metrics/cloudprovider.go

```diff
@@ -50,8 +56,23 @@ var methodDurationHistogramVec = prometheus.NewHistogramVec(
 	},
 )
 
+var (
+	errorsTotalCounter = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: metrics.Namespace,
+			Subsystem: "cloudprovider",
+			Name:      "errors_total",
+			Help:      "Total number of errors returned from CloudProvider calls.",
+		},
+		[]string{
+			metricLabelController,
```

Comment:
Done.

I also observe that the duration metrics in this package are an outlier compared to the rest of the project in that they invoke `.WithLabelValues()` (I am inferring this values-only configuration depends upon a well-ordered set of labels as defined in the `labelNames []string` parameter of `NewHistogramVec` in order to associate the values with specific labels)  instead of `.With()`. I prefer the more verbose `.With()` for code maintenance as we see the label-value assignment more explicitly. Do we want to consider updating the duration metrics implementation to use `.With()` as in-scope for this PR?

Reply:
Yeah, I think changing to `With()` makes sense. And I agree with you that I prefer the more verbose method as well. It's definitely less error-prone

Reply:
Done, I think this is ready for final review, PTAL, thanks!